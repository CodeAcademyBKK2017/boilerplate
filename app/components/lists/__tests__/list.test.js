import 'react-native';
import List from '../list.component';
import React from 'react';
import renderer from 'react-test-renderer';

// Note: test renderer must be required after react-native.
// import uuid from 'uuid';

import {shallow} from 'enzyme';

describe('List', () => {
  
  it('renders correctly', () => {
    const tree = renderer.create(
      <List />
    );
    expect(tree).toBeDefined();
  });

  it('List: renders correctly', () => {
    const tree = renderer.create(<List />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('List: _renderItems Function is will work', () => {
    const wrapper = shallow(<List/>);
    const instance = wrapper.instance();
    const param = {item: {title: 'test', content: 'test1', key: ''}};
    const tree = instance._renderItems(param);
    expect(tree).toMatchSnapshot();
  });

  it('List: _showPopup Function is will work', () => {
    const wrapper = shallow(<List/>);
    const instance = wrapper.instance();
    const param = {item: {title: 'test', content: 'test1', key: ''}};
    const tree = instance._showPopup(param);
    expect(tree).toMatchSnapshot();
  });
});
