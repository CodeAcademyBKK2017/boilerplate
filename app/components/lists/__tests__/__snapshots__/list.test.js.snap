// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List: _renderItems Function is will work 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "borderColor": "rgba(0, 0, 0, .10)",
      "borderTopWidth": 1,
      "padding": 5,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      test1
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`List List: _showPopup Function is will work 1`] = `[Function]`;

exports[`List List: renders correctly 1`] = `
<RCTScrollView
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "height": "10%",
      "padding": 10,
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
